types/mapbox-gl/index.d.ts

```diff
@@ -876,7 +881,11 @@ declare namespace mapboxgl {
     /**
      * Control
      */
-    export class Control extends Evented {}
+    export class Control extends Evented implements IControl {
+        onAdd(map: Map): HTMLElement;
+        onRemove(map: Map): any;
```

Comment:
Hi @xivk,

Looks good.

Just a small note, after reading the documentation I noticed that the `onRemove` method should return undefined (https://github.com/mapbox/mapbox-gl-js/blob/520a4ca3b9d260693f1e7f4af30d8b91c8f8bb08/src/ui/map.js#L2809)

Would't be a better idea to change the return value from `any` to `void` on this method then? I know it does not make a big difference. But at least now the user knows that any returned value will be ignored.

Reply:
@amxfonseca Is there even such a thing as returning "undefined" in TypeScript? And, i'd say it should return 'unknown', not 'any'. Many coding standards (lint rules) prohibit using 'any' these days.

Reply:
@dmytro-gokun `undefined` in typescript is just `void`, aka not returning anything at all. But if you explicitly return `undefined` value  on a function that has `void` as its return type the compiler won't complain.

So for this `void` would be the proper typing. `unknown` is just a better `any` where the compiler enforces you to cast or test for a type before using that value, this is not the case.

Reply:
Ah yes, I can do that!